### PR TITLE
Miscellaneous nRF patches

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Just a couple of patches we've had sitting in our fork that would be good to get upstream. These don't affect non-nRF SoCs.

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -429,7 +429,7 @@ config BOOT_SERIAL_DETECT_PIN
 	default 6 if BOARD_NRF9160DK_NRF9160
 	default 11 if BOARD_NRF52840DK_NRF52840
 	default 13 if BOARD_NRF52DK_NRF52832
-	default 23 if BOARD_NRF5340_DK_NRF5340_CPUAPP || BOARD_NRF5340_DK_NRF5340_CPUAPPNS
+	default 23 if BOARD_NRF5340PDK_NRF5340_CPUAPP || BOARD_NRF5340PDK_NRF5340_CPUAPPNS
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig:432]: And what with BOARD_NRF5340**DK**_NRF5340_CPUAPP ?

Comment (mbolivar-nordic) [reply boot/zephyr/Kconfig:432]: > And what with BOARD_NRF5340**DK**_NRF5340_CPUAPP ?

I don't understand your question. All I did was upstream a patch we had downstream.

Comment (ioannisg) [reply boot/zephyr/Kconfig:432]: I guess we need both, @nvlsianpu

Comment (nvlsianpu) [reply boot/zephyr/Kconfig:432]: yeap

**Pull request merged**